### PR TITLE
The Nextweb responsive design replica.

### PR DESCRIPTION
**Pull request by sinework**
This is a replica of thenextweb landing page made for the Microverse html/css track, responsive layouts section. Flex,floats and grids are used for the layout. Media queries were used to maintain its responsiveness.

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -0,0 +1,1085 @@
...
+  grid-gap: 6px;
+  grid-template-areas:
+    "main side-top "
+    "main side-bottom";
+  height: 50vh;
+  margin: 4px;
+}
+
+.main-pic {
+  /* background:  linear-gradient(rgba(196, 32, 32, 0.25)),url("./images/main-pic-big.png"); */
```
Comment (MuriloRoque) [inline css/style.css]: Please remove this commented piece of code.

File: css/style.css [hunk 2]
```diff
@@ -0,0 +1,1085 @@
...
+    grid-gap: 6px;
+    grid-template-areas:
+      "main side-top "
+      "main side-bottom";
+    height: 50vh;
+    margin: 4px 4px 5% 4px;
+  }
+
+  .main-pic {
+    /* background:  linear-gradient(rgba(196, 32, 32, 0.25)),url("./images/main-pic-big.png"); */
```
Comment (MuriloRoque) [inline css/style.css]: Please remove this commented piece of code.

File: css/style.css [hunk 3]
```diff
@@ -0,0 +1,1085 @@
...
+    grid-gap: 6px;
+    grid-template-areas:
+      "main"
+      "side-top"
+      "side-bottom";
+    margin: 4px 4px 5% 4px;
+  }
+
+  .main-pic {
+    /* background:  linear-gradient(rgba(196, 32, 32, 0.25)),url("./images/main-pic-big.png"); */
```
Comment (MuriloRoque) [inline css/style.css]: Please remove this commented piece of code.

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
## Status: Changes requested :red_circle: 

### Hello @sinework and @ajkacca457 :wave: 

### You have done an awesome job in this project, I could hardly notice any differences between yours and the original, it's really good, congrats :+1: 

### Let's get to my review, please check my comments below:

## Page

### Top part 
- **1024px**

- [x] Your TNW logo image is not loading here, please check this issue:

Screenshot from 2020-08-26 17-44-52
<br>
- **768px**

- [x] We are having some issues with these arrows in your navbar from 768px to around 854px, they should be aligned to the side of the link:

Screenshot from 2020-08-26 17-50-03

<br> 

- [x] Still, in the navbar, your bottom links shouldn't be centered, check the original:

***YOURS:***

Screenshot from 2020-08-26 17-51-56
<br>

***ORIGINAL:***

Screenshot from 2020-08-26 17-52-23

<br>

- [x] This image from the first article is also not loading in your medium and small screens:

Screenshot from 2020-08-26 17-55-37

### Latest Funding Section

- [x] You are missing some links here in this section, please add them:

***YOURS:***

Screenshot from 2020-08-26 17-58-48

<br>

***ORIGINAL:***

Screenshot from 2020-08-26 17-59-21

<br>

- [x] These texts shouldn't be centered, check the original:

***YOURS:***

Screenshot from 2020-08-26 18-02-21

<br>

***ORIGINAL:***

Screenshot from 2020-08-26 18-02-19
<br>

### Hard Fork, Tech, Apps, Gear, etc Section

- [x] You are missing an image beside the Latest part. Also, please make the bullets below that look more like the original:

***YOURS:***

Screenshot from 2020-08-26 18-05-49
<br>
***ORIGINAL:***

Screenshot from 2020-08-26 18-05-52
<br>

### Latest Deals Section

- [x] As we get to the 768px breakpoint, your text is too far away from the image to the left, check below:

Screenshot from 2020-08-26 18-09-47

### Footer

- [x] In all screens, you are missing the TNW logo, check the images below and please add it.

- [x] Your footer elements are not correctly aligned in all screen sizes. The social icons should be closer to each other when the screen is small and far from each other when the screen increases. Also, the links below the social icons need to be aligned properly. Please, check the screenshots below:

- **375px**

***YOURS:***

Screenshot from 2020-08-26 18-13-01
<br>
***ORIGINAL:***

Screenshot from 2020-08-26 18-13-03
<br>

- **768px and 1024px**

***YOURS:***

Screenshot from 2020-08-26 18-18-52
<br>

***ORIGINAL:***

Screenshot from 2020-08-26 18-18-50
<br>

## General

- [ ] Please, create classes for the elements below in your CSS and add them to the HTML, since you are using them multiple times, let's not make bad practices :+1: 

Screenshot from 2020-08-26 18-24-50
<br>

- [x] Don't forget to check my comments in your code below.

## Summary

- Your code is very good, your HTML is semantic and you are mostly using CSS best practices;
- The page is responsive and looks really good, we just need to fix some details;
- Your PR, README and commits are good too, great job guys :+1: 

## Kindly make the requested changes and submit for another review, please :pray: 

## I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

File: css/style.css (hunk 1 above)
Comment (ajkacca457) [reply css/style.css]: Hey, Thank you for your detailed constructive review. It is really tough to identify those mistakes by ourselves. Your review helped us in making productive changes in our project. Based on your review, We have fixed the black navigation bar links icon problem, We have updated the main tnw img location and it is now visible. We also have changed the text alignment of the yellow nav link to the left. Images beside the Latest part are also updated and fixed. We reduced the margin-right of the latest deal section in the smaller screen, We have fixed the footer navigation links alignment and added the twn logo at the bottom of the footer. Hopefully, you will find our project acceptable after the changes that we made. Thanks again for your inspiring review and for giving us the motivation to finish the project accurately.

File: css/style.css (hunk 2 above)
Comment (ajkacca457) [reply css/style.css]: Commented code removed.

File: css/style.css (hunk 3 above)
Comment (ajkacca457) [reply css/style.css]: Commented code removed.

File: css/style.css (hunk 1 above)
Comment (MuriloRoque) [reply css/style.css]: Thank you for your kind comment :heart:

I hope you guys can keep up the good work :muscle: 

Keep going and never stop, you will reach the :top: 

Have a very nice day :sun_with_face:

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## great job so far guys :clap: :+1: :100:, but you need to make some changes :hammer: :wrench: :toolbox:  to get this project approved :checkered_flag: 

- [x] Add some right margin to the image articles:

|*Clone*|*Original*|
|:---:|:---:|
| image | image |

- [x] your footer text is not orginized, make it look like the original:

|*Clone*|*Original*|
|:---:|:---:|
| image | image |

- [x] Fix the links below the latest funding title:

|*Clone*|*Original*|
|:---:|:---:|
| image | image |

- [x] align the title with the image article in the latest news section:

|*Clone*|*Original*|
|:---:|:---:|
| image | image |

- [x] On the mobile view the latest news is overlapping with the hero section.

- [x] These article titles should be on the bottom not on the centered:
image

### Please make all the required changes before you submit another code review.
## Happy Coding :tophat:

**Comment by ajkacca457** [comment]
@elmejdki Thank you for pointing out all the responsiveness issues. Your review was extremely helpful in fixing our web-design according to various screen sizes.  We have fixed the latest funding section and the links now look like the original one. We have added some margin to the latest deal images.  We made sure that the latest news section doesn't overlap with the main display. We increased the margin so that any overlapping doesn't occur. We also fixed the footer section for small devices and the text alignment of the footer section resembles the actual size. We aligned the heading to the bottom of each picture according to various device sizes. Thank you once again for your critical review. Hopefully, we were able to fix the issues according to the expectations. Have a great day.

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [x] there are big spaces between each section, please fix that

- [x] the latest funding cards decreases from 4 to 3 cards on medium and small screens

- [x] Your footer still doesn't match the website on small screens (notice how many lines the original have)

| Yours | Original |
| --- | --- |
| image | image |

- [x] Your latest funding links still doesn't match the website on small screens

image





### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by ajkacca457** [comment]
@Spayco  Thank you for your review. We have fixed most of the issues that you mentioned. We have aligned the footer, reduced the gap between sections, styled the funding section's link according to the actual website, and reduced one element from funding cards.  Hopefully, It will work according to the expectation. Thanks again for your valuable feedback.

File: css/style.css [hunk 4]
```diff
@@ -0,0 +1,1544 @@
...
+    margin-bottom: 10px;
+  }
+
+  #br {
+    display: none;
+  }
+
+  #br-first {
+    display: none;
+  }
```
Comment (kevcap) [inline css/style.css]: - [x] OPTIONAL ➡ Those two selelectors do the same thing, yet they're separated plust you're using ID instead classes. Kindly fix that

File: css/style.css [hunk 5]
```diff
@@ -0,0 +1,1544 @@
...
+  margin-right: auto;
+}
+
+.title-side {
+  margin: 0 6px;
+}
+
+.p-holder {
+  display: flex;
+}
```
Comment (kevcap) [inline css/style.css]: - [x] OPTIONAL ➡ `display: flex` is repated 17 times troughout your code. I recommend you guys to create a `.d-flex { display: flex }` class to handle that rule, in order to keep your code DRY.

File: css/style.css [hunk 6]
```diff
@@ -0,0 +1,1544 @@
...
+  background-color: purple;
+}
+
+#searchbtn {
+  transform: rotate(90deg);
+  color: #aaa;
+}
+
+#hamburger {
+  display: none;
```
Comment (kevcap) [inline css/style.css:251]: - [x] OPTIONAL ➡ `display: none` is repeated 20+ times throughout your code. I recommend you guys to create a `.d-none {display: none}` class to handle that rule, in order to keep your DRY code.

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

## Hi team, good job on making the previously required changes!
Your page meets most of the requirements  👏 
You just have to implement some minor changes in order to move on
 
## [REQUIRED CHANGES]
- [x] Your latest foundigs still don't match in medium screens. Kindly fix that guys! 💪
<table>
<tr>
 <td></td>
 <td></td>
 </tr>
</table>

## [OPTIONAL CHANGES]
⚠ These are mandatory changes, but since It wasn't pointed out earlier, I'll mark them as **[OPTIONAL]**:
- [x] Kindly remember to remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png) from Pathwright)
Take a look at this example:
image
Your HTML will be something like this: 
image
Apply the same logic above in **ALL** those repeated rules listed below:
image

- [x] Some issues with the layout: 
#### Small screens issues:
<table>
 <tr>
 <th>Yours [SMALL](375px)</th>
 <th>Original [SMALL](375px)</th>
 </tr>
 <tr>
 <td></td>
 <td></td>
 </tr>
 <tr>
 <td></td>
 <td></td>
 </tr>
 <tr>
 <td></td>
 <td></td>
 </tr>
</table>

#### Medium screens issues:
<table>
 <tr>
 <th>Yours [MEDIUM](768px)</th>
 <th>Original [MEDIUM](768px)</th>
 </tr>
 <tr>
 <td></td>
 <td></td>
</tr>
<tr>
 <td></td>
 <td></td>
 </tr>
</table>

#### Large screens issues:
<table>
 <tr>
 <th>Yours [LARGE](1024px)</th>
 <th>Original [LARGE](1024px)</th>
 </tr>
 <tr>
 <td></td>
 <td></td>
</tr>
</table>



## Please make the required changes and resubmit for a new code review once you are ready💪

## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

  
<sub>*We're here to help you, to ensure that you're using the best practices and getting you ready for the job market. In real-world, you'll have a lot of changes requests for your real clients, it's good to get used to it.</sub>

**Comment by ajkacca457** [comment]
@kevincappuccino  thank you for your further review. We were facing a layout problem because of an element was not aligned properly in our previous version of the website. We have modified the latest news section and the display section with flex property instead of a complex grid area layout. Now both of the section doesn't overlap each other in any devices. We have also fixed the funding links alignment on the small screen. We have also reduced the font-weight that you mentioned.  And lastly, we have tried to eliminate any repeated code from our CSS files. We have tried to strictly follow the dry principle this time. Thank you again for pointing out the obvious mistakes. Hopefully, Now the project will meet the requirements.

File: css/style.css [hunk 7]
```diff
@@ -0,0 +1,1255 @@
...
+
+  /* logo sizing */
+
+  .logo-TNW img {
+    display: none;
+  }
+
+  .logo-TNW p {
+    display: none;
+  }
```
Comment (andresporras3423) [inline css/style.css]: Group selectors with the same style definition

File: css/style.css [hunk 8]
```diff
@@ -0,0 +1,1255 @@
...
+    border-bottom: 1px rgba(84, 84, 84, 0.27) solid;
+    border-top: 1px rgba(84, 84, 84, 0.27) solid;
+    padding-bottom: 10px;
+    padding-left: 15px;
+    padding-top: 15px;
+    width: 35px;
+    font-size: 110%;
+  }
+
+  #hamburger {
```
Comment (andresporras3423) [inline css/style.css]: Don't overuse ID selectors

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Sine Shiferaw & Avi Karmaker** 
Good job with the improvements for this project :muscle:  :muscle:
Your page meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:
- [x] Don't overuse ID Selectors
- [x] Don't repeat yourself, in order to minimize code you can group selectors with the same style definition.
- [x]  Check the footer of your page, you have missed some part of the text, compare with the original:

| Yours | Original |
|:---:|:---:|
|image | image |

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

File: css/style.css (hunk 7 above)
Comment (sinework) [reply css/style.css]: We have tried to group selectors with the same value. Limited use of ID selectors to just one and instead used classes and we have also added the missing footer text.

**Review by peterrobert — CHANGES_REQUESTED** [review]
### HELLO THERE 👋 .  SOME FEW CHANGES REQUIRED ♻️ 
### VERY NICE WORK ON THIS PROJECT. HOWEVER THERE IS ONE MORE CHANGE TO BE MADE.

- [x] Looking at your project, the logo image link appears broken. Kindly fix this.

**Comment by sinework** [comment]
The logo appears to be working fine all screen sizes on multiple screen sizes. Please check the live link.

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work, team! 🖥️ 👀 ✨ 
## However, some changes are still required... 🧰 🔨 🔧 

- When viewing the `index.html` file from your `feature` branch through githack, the logo does appear broken, as stated by the previous TSE:

https://rawcdn.githack.com/sinework/The-Next-Web/58df2fcf2e2810c9718984977a9bae0942fefca3/index.html

Your [Netlify App](https://pensive-booth-3ecd98.netlify.app/) seems to be working. Just make sure that your `index.html` also works properly, as implied by the previous TSE's request.

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Comment by sinework** [comment]
Thank you for pointing out where the problem was. We haven't been using raw githack so we weren't able to see the broken link before. We have now fixed the path issue that was creating the problem.

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 

good job implementing the changes, you still one more thing though

## Required Changes ♻️ 

* [x] Files not related to the project should not be committed and added to .gitignore ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)), you have many images that you're not using and should be deleted.

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Comment by sinework** [comment]
Deleted images that are not related to the project.

**Review by voscarmv — APPROVED** [review]
# Impressive work, team! 🖥️ 👀 ✨ 

- Great work implementing all the requested changes.

# Project Status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**